sstmac/common/event_scheduler.cc

```diff
@@ -153,6 +153,16 @@ Component::setup()
 #endif
 }
 
+Component::Component(uint32_t cid, SSTMAC_MAYBE_UNUSED SST::Params&  params) :
+#if SSTMAC_INTEGRATED_SST_CORE
+   SSTIntegratedComponent(params, cid),
+   EventScheduler(cid, this)
```

Comment:
I will need to understand this better. The (undocumented) intent is that EventScheduler is never used with core. I would prefer to avoid this inheritance pattern.  What was the reason for doing this? (so I can think about the best way)

Reply:
I didn't change this.  I just move this function from the header to the .cc file.  I moved it because event_scheduler is one of the files we spend the most time in compilation wise. So I can't tell you why it initializes the way it does.